### PR TITLE
EVM documentation improvements

### PR DESCRIPTION
**Pull request by dssei**
Adding rust docs for EVM related stucts/functions

### CODE REVIEW

**Comment by dssei** [comment]
@ARitz-Cracker please have a look.

File: packages/sei-cosmwasm/src/querier.rs [hunk 1]
```diff
@@ -187,7 +206,19 @@ impl<'a> SeiQuerier<'a> {
         self.querier.query(&request)
     }
 
-    // returns base64-encoded bytes
+    /// Query to get hex payload for the ERC-20 `transfer` function
+    ///
+    /// # Arguments
+    /// * `recipient` - Sei recipient address.
```
[the remark this answers is not in the document]
Comment (codchen) [reply packages/sei-cosmwasm/src/querier.rs:212]: cosmos

**Review by codchen — APPROVED** [review]

**Review by besated — APPROVED** [review]

File: packages/sei-cosmwasm/src/querier.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (dssei) [reply packages/sei-cosmwasm/src/querier.rs:212]: Yeah, I guess through the docs I was referring to x0 addresses as `EVM` and to bech32 as `Sei`. But Sei is probably confusing as we support both.

**Review by ARitz-Cracker — CHANGES_REQUESTED** [review]
Very much appreciated! Though I still wholly believe that more explicit language would benefit newcomers. Most of my comments can apply to other sections of this PR.

File: packages/sei-cosmwasm/src/msg.rs [hunk 2]
```diff
@@ -41,13 +41,25 @@ pub enum SeiMsg {
...
+    /// The EVM (Solidity) contract `msg.sender` in this case will be the callers address.
     DelegateCallEvm {
+        /// The address of the EVM contract to call
         to: String,
+        /// Base64 encoded data to pass to the contract
         data: String, // base64 encoded
     },
+    /// Calls EVM contract deployed at `to` address with specified `value` and `data`.
+    /// The from address is the contract address of the contract executing the call.
+    /// The EVM (Solidity) contract `msg.sender` in this case will be the CW contract address.
```
[the remark this answers is not in the document]
Comment (dssei) [reply packages/sei-cosmwasm/src/msg.rs]: Done

File: packages/sei-cosmwasm/src/msg.rs [hunk 3]
```diff
@@ -41,13 +41,25 @@ pub enum SeiMsg {
     SetMetadata {
         metadata: Metadata,
     },
+    /// Calls EVM contract deployed  at `to` address with the given `data`.
+    /// The from address is the callers address.
```
[the remark this answers is not in the document]
Comment (dssei) [reply packages/sei-cosmwasm/src/msg.rs]: Replaced with suggested text

**Comment by dssei** [comment]
> Very much appreciated! Though I still wholly believe that more explicit language would benefit newcomers. Most of my comments can apply to other sections of this PR.

@ARitz-Cracker thanks for reviewing. Will be addressing most comments shortly

File: packages/sei-cosmwasm/src/msg.rs [hunk 4]
```diff
@@ -41,13 +41,25 @@ pub enum SeiMsg {
...
+    /// Please note that the CW contract has to be in allow list in order to execute delegate call.
+    /// The EVM (Solidity) contract `msg.sender` in this case will be the callers address.
     DelegateCallEvm {
+        /// The address of the EVM contract to call
         to: String,
+        /// Base64 encoded data to pass to the contract
         data: String, // base64 encoded
```
[the remark this answers is not in the document]
Comment (codchen) [reply packages/sei-cosmwasm/src/msg.rs:52]: yes all bytes will be transferred in base64 format

File: packages/sei-cosmwasm/src/msg.rs [hunk 5]
```diff
@@ -41,13 +41,25 @@ pub enum SeiMsg {
     SetMetadata {
         metadata: Metadata,
     },
+    /// Calls EVM contract deployed  at `to` address with the given `data`.
+    /// The from address is the callers address.
+    /// Please note that the CW contract has to be in allow list in order to execute delegate call.
```
[the remark this answers is not in the document]
Comment (codchen) [reply packages/sei-cosmwasm/src/msg.rs:46]: it refers to any CW code hash in this list https://github.com/sei-protocol/sei-chain/blob/seiv2/x/evm/types/params.go#L142 (currently only the pointer contracts).

File: packages/sei-cosmwasm/src/msg.rs [hunk 6]
```diff
@@ -41,13 +41,25 @@ pub enum SeiMsg {
...
+    /// The EVM (Solidity) contract `msg.sender` in this case will be the callers address.
     DelegateCallEvm {
+        /// The address of the EVM contract to call
         to: String,
```
[the remark this answers is not in the document]
Comment (codchen) [reply packages/sei-cosmwasm/src/msg.rs:50]: no it doesn't

File: packages/sei-cosmwasm/src/msg.rs [hunk 7]
```diff
@@ -41,13 +41,25 @@ pub enum SeiMsg {
     SetMetadata {
         metadata: Metadata,
     },
+    /// Calls EVM contract deployed  at `to` address with the given `data`.
+    /// The from address is the callers address.
+    /// Please note that the CW contract has to be in allow list in order to execute delegate call.
+    /// The EVM (Solidity) contract `msg.sender` in this case will be the callers address.
```
[the remark this answers is not in the document]
Comment (codchen) [reply packages/sei-cosmwasm/src/msg.rs:47]: @besated do you know if we already have a sei doc page for address mapping? If not I can help with writing up one (not sure where the gitbook repo is)

[the remark this answers is not in the document]
Comment (besated) [reply packages/sei-cosmwasm/src/msg.rs:49]: we have this section in the docs - https://v2.docs.sei.io/setting-up-a-wallet#advanced

if you want to add more details, you can create a pr in the `sei-docs` repo

**Comment by dssei** [comment]
@ARitz-Cracker addressed the most comments and will merge it. We can add more improvements in subsequent PRs

**Pull request merged**